### PR TITLE
Support example tag on structs

### LINKED ISSUE #32: Example tag
How define example values for parameters?

### PR DESCRIPTION
**Pull request by nikicc**
Fixes #32 

This PR introduces the support for example tags on structs that can be used to generate examples in Open API.

### CODE REVIEW

File: README.md
Comment (wI2L) [inline README.md:193]: s/Open API/OpenAPI

File: openapi/generator.go
Comment (wI2L) [inline openapi/generator.go:763]: `sf.Tag.Get` is good enough in place of `Lookup` since we can juste skip if the value is empty, but we should use `strings.TrimSpace` before checking that `e != ""`.

File: openapi/generator.go
Comment (wI2L) [inline openapi/generator.go:764]: The error should be dealt-with if non-nil.
If suggest that we silently ignore the error but save it with`g.error(&FieldError{...`. See some examples in `generator.go`. Those errors can be retrived with `fizz.Errors`.

File: openapi/generator.go
Comment (wI2L) [inline openapi/generator.go:1175]: While it may sound ok, the `schemaType` received here is a literal string value that emanate from the `schema.Type` field, which is intended for the OpenAPI spec, it doesn't represent a native Go type.
Instead, we should take a `reflect.Type`. In `newSchemaFromStructField`, you can pass the value of `sf.Type` which represents the reflection type of the struct field.

File: openapi/generator.go
Comment (wI2L) [inline openapi/generator.go:1180]: `strconv.Atoi` is too naive here. What about `uint8` fields that would provide an oveflowing example value ? We need to use `strconv.ParseInt` or `strconv.ParseUint` with the proper `bitSize`. If an error occures, we return it and leave the caller decide what to do (ignore the example value and collect the error).

**Review by wI2L — CHANGES_REQUESTED** [review]
Thank you for your first contribution.
The basis of changes is solid, but the implementation requires a few fixes.

**Review by nikicc — COMMENTED** [review]
@wI2L thanks for a prompt review. I've made the changes and this is ready for another check.

~Also, I think the CodeCov report was not updated after all the changes. Is this expected?~ I think all is good, just pushed one more test to increase the coverage.

File: openapi/generator.go
Comment (wI2L) [inline openapi/generator.go:763]: `strings.TrimSpace` is only used in the condition check here.
Instead, write `strings.TrimSpace(sf.Tag.Get("example"))`.

File: openapi/generator.go
Comment (wI2L) [inline openapi/generator.go:766]: the param `e` isn't the field's name, but the raw value being parsed.
Just use smthing like `example value %q cannot be parsed: %s`.

File: openapi/generator.go
Comment (wI2L) [inline openapi/generator.go:1182]: Not a huge deal, but for consistency accros the library, please rename the param to `typ` or `t`.

File: openapi/generator.go
Comment (wI2L) [inline openapi/generator.go:1190]: ~~Should also handle `reflect.Uintptr`.~~
Edit: we don't handle it already in  the library, so no point I guess.

File: openapi/generator.go
Comment (wI2L) [inline openapi/generator.go:1195]: I'd remove the function name in the error, it doesn't bring any value. Return smthing like `unsupported type: %s`with `fmt.Errorf`, and eventually in the parent give more context about the error, like "couldn't parse example value of field...".

File: openapi/generator_test.go
Comment (wI2L) [inline openapi/generator_test.go:303]: Testing that the error message is not empty is good, but if we're doing that, we could at least test for the presence of other fields as-well (fiedl name, type...) if that's relevant to render a proper/understandable error message.

File: openapi/generator_test.go
Comment (wI2L) [inline openapi/generator_test.go:624]: s/schema/typ ?

**Review by wI2L — CHANGES_REQUESTED** [review]
Small quirks remaining there and there, should be ready to merge after fixes.

**Comment by nikicc** [comment]
@wI2L sorry, I re-requested review by mistake 🙈  I'm still working on the changes, will let you know once it's ready again.

File: openapi/generator.go
Comment (nikicc) [reply openapi/generator.go]: Changed. Also reworded the message in parent a bit to give the field name as well.

File: openapi/generator_test.go
Comment (nikicc) [reply openapi/generator_test.go]: I was just copying this from the test above 🙈  But yes, we should do more. Changed, so we now assert the error message.

File: openapi/generator_test.go
Comment (nikicc) [reply openapi/generator_test.go]: 👍

**Review by nikicc — COMMENTED** [review]
@wI2L now this is ready. Please, check again and let me know if you find anything else 😊

File: openapi/generator_test.go
Comment (wI2L) [inline openapi/generator_test.go:303]: I wasn't thinking of that kind of test, perhaps I wasn't clear enough.
What I had in mind was a type cast f the error to `fizz.FieldError`, then check if the fields `Name`, `Typ` values match what we expect to see. The format of the error itself isn't that relevant.

**Review by wI2L — CHANGES_REQUESTED** [review]
One last test to fix, and we'll be done.

File: openapi/generator_test.go
Comment (wI2L) [reply openapi/generator_test.go:303]: However, we need the previous `assert` that checked if the return of `Error()` isn't empty.

Comment (nikicc) [reply openapi/generator_test.go]: Ohhhh ... I think I get now what you meant. Just pushed, please check 🙏

**Review by wI2L — APPROVED** [review]

**Comment by wI2L** [comment]
Thank you for your contribution @nikicc.

**Comment by nikicc** [comment]
@wI2L thanks for being so responsive 🚀  I've set the signature now, but you'll need to re-approve, sorry 🙈 

Also another quick question: what's the releasing strategy? Could we release this after being merged?

**Comment by wI2L** [comment]
@nikicc Yes I will push a new tag after I merge.

**Review by wI2L — APPROVED** [review]

**Comment by wI2L** [comment]
Release v0.14.0 published @nikicc.

**Comment by nikicc** [comment]
@wI2L yeeey 🎉   Thanks!

**Pull request merged**
